common/src/main/java/com/wynntils/screens/WynntilsQuestBookScreen.java

```diff
@@ -0,0 +1,513 @@
+                        Texture.QUEST_BOOK_BACKGROUND.height() * 1.3f - 6f,
+                        0,
+                        CommonColors.YELLOW,
+                        HorizontalAlignment.Center,
+                        FontRenderer.TextShadow.NORMAL);
+        poseStack.popPose();
+    }
+
+    private List<Component> getTooltipLinesForQuest(QuestInfo questInfo) {
+        List<Component> tooltipLines = new ArrayList<>() {
+            {
+                add(new TextComponent(questInfo.getName())
+                        .withStyle(ChatFormatting.BOLD)
+                        .withStyle(ChatFormatting.WHITE));
+                add(questInfo.getStatus().getComponent());
+                add(new TextComponent(""));
+                add((CharacterManager.getCharacterInfo().getLevel() >= questInfo.getLevel()
+                                ? new TextComponent("✔").withStyle(ChatFormatting.GREEN)
+                                : new TextComponent("✖").withStyle(ChatFormatting.RED))
+                        .append(new TextComponent(" Combat Lv. Min: ").withStyle(ChatFormatting.GRAY))
+                        .append(new TextComponent(String.valueOf(questInfo.getLevel()))
+                                .withStyle(ChatFormatting.WHITE)));
+            }
+        };
+
+        for (Pair<String, Integer> additionalRequirement : questInfo.getAdditionalRequirements()) {
+            MutableComponent base = CharacterManager.getCharacterInfo()
+                                    .getProfessionInfo()
+                                    .getLevel(ProfessionInfo.ProfessionType.valueOf(additionalRequirement.a))
+                            >= additionalRequirement.b
+                    ? new TextComponent("✔ ").withStyle(ChatFormatting.GREEN)
+                    : new TextComponent("✖ ").withStyle(ChatFormatting.RED);
+
+            tooltipLines.add(base.append(new TextComponent(additionalRequirement.a + " Lv. Min: ")
+                    .withStyle(ChatFormatting.GRAY)
+                    .append(new TextComponent(String.valueOf(additionalRequirement.b))
+                            .withStyle(ChatFormatting.WHITE))));
+        }
+
+        tooltipLines.add(new TextComponent("-")
+                .withStyle(ChatFormatting.GREEN)
+                .append(new TextComponent(" Length: ").withStyle(ChatFormatting.GRAY))
+                .append(new TextComponent(StringUtils.capitalizeFirst(
+                                questInfo.getLength().toString().toLowerCase(Locale.ROOT)))
+                        .withStyle(ChatFormatting.WHITE)));
+        tooltipLines.add(new TextComponent(""));
+
+        if (questInfo.getStatus() != QuestStatus.COMPLETED) {
+            tooltipLines.add(new TextComponent(questInfo.getNextTask()).withStyle(ChatFormatting.GRAY));
+            tooltipLines.add(new TextComponent(""));
+        }
+
+        if (questInfo.getStatus() != QuestStatus.CANNOT_START) {
```

Comment:
... at least up to this point. That would allow us to potentially reuse the code that generates the "actual" quest lore. The part below is more specifically tied to this Screen.